src/Microsoft.AspNetCore.Sockets/ConnectionMetadata.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    internal class ConnectionMetadata : IDictionary<object, object>
```

Comment:
This is pretty much a clone from https://github.com/aspnet/HttpAbstractions/blob/a79aea3bf379ef0752567474d9490279c6818ab1/src/Microsoft.AspNetCore.Http/Internal/ItemsDictionary.cs